File: C-sharp/Labwork 5/MainFlow/Capturer/PersonCapturer.cs
namespace Labwork_5.MainFlow.Capturer
{
    public class PersonCapturer
    {
        private static int CaptureAge()
        {
            int age = default;
            bool exceptionIsThrown = true;

            while (exceptionIsThrown)
            {
                System.Console.Write("Please, enter the celebrant's age: ");
                exceptionIsThrown = false;

                try
                {
                    age = int.Parse(Console.ReadLine());
                    Validator.ValidateAge(age);
                }
                catch (FormatException)
                {
                    System.Console.WriteLine("The entered value isn't a number");
                    exceptionIsThrown = true;
                }
                catch (ArgumentOutOfRangeException ex)
                {
                    System.Console.WriteLine(ex.Message);
                    exceptionIsThrown = true;
                }
            }

            return age;
        }

        public static void CapturePerson(Event activity)
        {
            PersonModel person = new PersonModel();
            bool exceptionIsThrown = true;
            
            while (exceptionIsThrown)
            {
                exceptionIsThrown = false;

                try
                {
                    System.Console.Write("Please, enter the first name: ");
                    person.FirstName = Console.ReadLine();

                    System.Console.Write("Please, enter the second name: ");
                    person.SecondName = Console.ReadLine();

                    System.Console.Write("Please, enter the patronymic: ");
                    person.Patronymic = Console.ReadLine();

                    if (activity is Birthday birthday)
                    {
                        birthday.CelebrantsAge = CaptureAge();
                        birthday.Celebrant = person;
                    }
                    else if (activity is Meeting meeting)
                    {
                        meeting.PersonToMeet = person;
                    }
                }
                catch (ArgumentException ex)
                {
                    System.Console.WriteLine(ex.Message);
                    exceptionIsThrown = true;
                }
            }
        }
    }
}

File: Python/Labwork 5/Capturer/person_capturer.py
from events import *
from person_model import *


class PersonCapturer:
    @staticmethod
    def __capture_age(birthday):
        age = 0
        exception_is_thrown = True

        while exception_is_thrown:
            exception_is_thrown = False
            age = int(input("Please, enter the celebrant's age: "))

            try:
                Birthday(birthday).set_age(age)
            except TypeError:
                print("The entered value isn't a number")
                exception_is_thrown = True
            except ValueError as ve:
                print(ve)
                exception_is_thrown = True

        return age

    @staticmethod
    def capture_person(activity):
        person = PersonModel(' ', ' ', ' ')
        exception_is_thrown = True

        while exception_is_thrown:
            exception_is_thrown = False

            try:
                person.set_first_name(input("Please, enter the first name: "))
                person.set_second_name(input("Please, enter the second name: "))
                person.set_patronymic(input("Please, enter the patronymic: "))

                if isinstance(activity, Birthday):
                    PersonCapturer.__capture_age(activity)
                    activity.set_celebrant(person)
                elif isinstance(activity, Meeting):
                    activity.set_person(person)
            except ValueError as ve:
                print(ve)
                exception_is_thrown = True

        return person